src/dataflow/find_command_result.rs

```diff
@@ -0,0 +1,182 @@
+                            )
+                        }
+                        Err(e) => FindCommandResult::Error(e),
+                    }
+                }
+            }
+            Err(e) => FindCommandResult::Error(e),
+        }
+    }
+
+    fn find_command_in_lockfile<S: AsRef<str>>(command_name: S, lockfile: Lockfile) -> Self {
+        match lockfile.get_command(command_name.as_ref()) {
+            Ok(lockfile_command) => {
+                match lockfile.get_module(
+                    &lockfile_command.package_name,
+                    &lockfile_command.package_version,
+                    &lockfile_command.module,
+                ) {
+                    Ok(lockfile_module) => {
+                        let path = PathBuf::from(&lockfile_module.entry);
+                        FindCommandResult::CommandFound(
+                            path,
+                            lockfile_command.main_args.clone(),
+                            lockfile_module.name.clone(),
+                        )
+                    }
+                    Err(_e) => {
+                        FindCommandResult::CommandNotFound(command_name.as_ref().to_string())
+                    }
+                }
+            }
+            Err(_e) => FindCommandResult::CommandNotFound(command_name.as_ref().to_string()),
+        }
+    }
+
+    pub fn find_command_in_directory<P: AsRef<Path>, S: AsRef<str>>(
+        directory: P,
+        command_name: S,
+    ) -> Self {
+        let manifest_result = ManifestResult::find_in_directory(&directory);
+        let lockfile_result = LockfileResult::find_in_directory(&directory);
+        match (manifest_result, lockfile_result) {
+            (ManifestResult::ManifestError(e), _) => return FindCommandResult::Error(e.into()),
+            (_, LockfileResult::LockfileError(e)) => return FindCommandResult::Error(e.into()),
+            (ManifestResult::NoManifest, LockfileResult::NoLockfile) => {} // continue
+            (ManifestResult::NoManifest, LockfileResult::Lockfile(l)) => {
+                return Self::find_command_in_lockfile(command_name, l)
+            }
+            (ManifestResult::Manifest(_m), LockfileResult::NoLockfile) => {
+                panic!("Manifest exists, but lockfile not found!")
+            }
+            (ManifestResult::Manifest(m), LockfileResult::Lockfile(l)) => {
+                return Self::find_command_in_manifest_and_lockfile(command_name, m, l);
+            }
+        };
+        FindCommandResult::CommandNotFound(command_name.as_ref().to_string())
+    }
+}
+
+/// Get a command from anywhere, where anywhere is the set of packages in the local lockfile and the global lockfile.
+/// A flag indicating global run is also returned. Commands are found in local lockfile first.
+pub fn get_command_from_anywhere<S: AsRef<str>>(
+    command_name: S,
+) -> Result<(PathBuf, Option<String>, String, bool), Error> {
```

Comment:
I think a `struct` with named fields for the success case here would improve readability